### PR TITLE
Update README for Java API client v2

### PR DESCRIPTION
**Pull request by ElPicador**

### CODE REVIEW

File: include/templates/README.md [hunk 1]
```diff
@@ -120,7 +128,8 @@ Table of Contents
 1. [Keep-alive](#keep-alive)
 1. [Debugging](#debugging)
 <% end -%>
-<% if scala? -%>1. [Philosophy of the scala client](#philosophy)<% end -%>
+<% if scala? -%>1. [Philosophy of the scala client](#philosophy)<% end %>
```
Comment (CBaptiste) [inline include/templates/README.md]: should be <% end -%>

File: include/templates/README.md [hunk 2]
```diff
@@ -120,7 +128,8 @@ Table of Contents
...
+<% if java2? -%>1. [Philosophy of the java client](#philosophy)<% end %>
```
Comment (CBaptiste) [inline include/templates/README.md]: same

File: include/templates/README.md [hunk 3]
```diff
@@ -500,6 +509,22 @@ client.execute {
...
+The v2 of the api client, uses a builder to create the APIClient object. If you are on a regular JVM (not android, not Google App Engine), use the `ApacheAPIClientBuilder`, if you are on Google App Engine use the `AppEngineAPIClientBuilder`.
+As for Android, please use the [Android API Client](https://github.com/algolia/algoliasearch-client-android)
+
+
+JSON & Jackson2
+--------------
+All the serialization/deserialization is done with Jackson2. You can add your custom ObjectMapper with the method `setObjectMapper` of the builder.
+Changing it might result in unexpected result. You can find the one used in the interface `com.algolia.search.Defaults.DEFAULT_OBJECT_MAPPER`.
+
+<% end %>
```
Comment (CBaptiste) [inline include/templates/README.md]: should be <% end -%>

File: include/templates/README.md (hunk 1 above)
Comment (ElPicador) [inline include/templates/README.md]: if I remove I do not have a cariage return on the scala README so `Philosophy of the scala client` and `Guides & Tutorials]` are on the same line

File: include/templates/README.md (hunk 2 above)
Comment (ElPicador) [inline include/templates/README.md]: same reason

File: include/templates/README.md [hunk 4]
```diff
@@ -1334,8 +1361,9 @@ Specify the list of approximation that should be considered as an exact match in
     <% end %>
   <% end %>
 
-
-
+<% if java2? -%>
```
Comment (CBaptiste) [inline include/templates/README.md]: should be <% if java2? %>

File: include/templates/README.md [hunk 5]
```diff
@@ -1334,8 +1361,9 @@ Specify the list of approximation that should be considered as an exact match in
...
+To add a custom setting for an index, simply extend the class `IndexSettings` and add your getter/setter. The new settings will be automatically serialized by jackson.
+<% end -%>
```
Comment (CBaptiste) [inline include/templates/README.md]: should be <% end %>

File: include/templates/README.md [hunk 6]
```diff
@@ -1334,8 +1361,9 @@ Specify the list of approximation that should be considered as an exact match in
     <% end %>
   <% end %>
 
-
-
+<% if java2? %>
+To add a custom setting for an index, simply extend the class `IndexSettings` and add your getter/setter. The new settings will be automatically serialized by jackson.
+<% end %>
```
Comment (CBaptiste) [inline include/templates/README.md]: can you add a line break after <% end %> please ? should be the last update

**Pull request merged**